src/main/java/io/cloudsoft/dbaccess/client/AbstractDatabaseAccessClient.java

```diff
@@ -106,142 +146,191 @@ public String getJdbcUrl(String user, String pass) {
+            return TemplateProcessor.processTemplateContents(s, MutableMap.of(
+                "db", getDatabase(),
+                "user", username,
+                "pass", password));
+        }
+        
+        public List<String> parseSqlScript(String in) throws SQLException {
+            // based on http://stackoverflow.com/questions/1497569/how-to-execute-sql-script-file-using-jdbc
+            Scanner s = new Scanner(in);
+            s.useDelimiter("(;(\r)?\n)|(--\n)");
+            List<String> result = MutableList.of();
+            while (s.hasNext()) {
+                String line = s.next();
+                if (line.startsWith("/*!") && line.endsWith("*/")) {
+                    // unescape C-style comments
+                    int i = line.indexOf(' ');
+                    line = line.substring(i + 1, line.length() - " */".length());
+                }
+                if (!Strings.isBlank(line)) {
+                    result.add(line);
+                }
             }
-        } catch (Exception e) {
-            Exceptions.propagateIfFatal(e);
+            s.close();
+            return result;
         }
-    }
-
-    public void close(Connection connection) {
-        try {
-            if (connection != null) {
-                connection.close();
+        
+        public void runStatements(String description, List<String> statements) {
+            if (statements==null || statements.isEmpty()) return;
+            try (Statement statement = connection.createStatement()) {
+                for (String s : statements) {
+                    for (String si: parseSqlScript(templated(s))) {
+                        LOG.debug("Executing: "+si);
+                        statement.execute(si);
+                    }
+                }
+            } catch (SQLException e) {
+                throw Exceptions.propagate("Error executing SQL for "+description, e);
+            }            
+        }
+        
+        public void createUser() {
+            runStatements("create user", getCreateUserStatements());
+        }
+    
+        private void grantPermissions() {
+            runStatements("grant permissions", getGrantPermissionsStatements());
+        }
+    
+        private void grantCustomAccessScript() {
+            if (Strings.isBlank(accessScript)) return;
+            runStatements("grant custom access script", MutableList.<String>builder().addAll(accessScript.split("\n")).build());
+        }
+        
+        private void deleteUser() {
+            runStatements("delete user", getDeleteUserStatements());
+        }
+        
+        protected List<String> getGrantPermissionsStatements() {
+            List<String> permissionStatements = Lists.newArrayList();
+            
+            if (AccessModes.CUSTOM == accessMode) return permissionStatements;
+            
+            // at a minimum should have SELECT permission
+            permissionStatements.addAll(getGrantSelectPermissionStatements());
```

Comment:
because we treat RO separately we shouldn't lump it with other `Permission`s imo; i will update docs however